docs: endorsement lifecycle management API (#74)
veraison/docs: api/endorsement-provisioning/README.md

    "expiry": "2030-10-12T07:20:50.52Z"
  }
```

# Endorsement Lifecycle Management (ELM) Interface

This interface can be used for activating/deactivating endorsements provisioned

> to the verifier. It allows using CoSERV base types to define an ELM query that 

Reviewer: ```suggestion
to the verifier. It uses the CoSERV base types to define an ELM query that 
```

Reviewer 2: Thanks for the suggestions. I've addressed all of them.

Reviewer: thank you!

can be used to select endorsements using either (stateful) environments or one or
more RIM identifiers.

## Query Format

Following is the CDDL format for an ELM query:
```
elm-query = {
 ( environment-query // rim-query )
}

environment-query = (
  &(profile: 0) => coserv.profile
  &(artifact-type: 1) => coserv.artifact-type
  &(environment-selector: 2) => coserv.environment-selector-map
)

rim-query = (
  &(rim-selector: 3) => [ + coserv.rim-selector-id ]
)
```

## Examples

### Deactivate keys
Use the `environment-query` variant to deactivate two keys by `instance-id`.

#### Request

> ```

Reviewer: ```suggestion
```http
```

POST /endorsement-provisioning/v1/deactivate
Host: veraison.example
Content-Type: application/vnd.veraison.elm-v1+cbor

-- body in EDN

{
  / profile /              0: "tag:arm.com,2025/example-profile",

>   / artifact-type /        1: 1 / trust-anchors /

Reviewer: ```suggestion
  / artifact-type /        1: 1, / trust-anchors /
```

  / environment-selector / 2: {
    / instance / 1: [ 
      [ 550( h'01 ...' ) ],
      [ 550( h'01 ...' ) ]
    ]

>   },

Reviewer: ```suggestion
  }
```

}
```

#### Response

> ```

Reviewer: ```suggestion
```http
```

HTTP/1.1 204 No Content
```

### Activate RIMs
Use the `rim-query` variant to activate two CoRIMs and one CoMID.

#### Request

> ```

Reviewer: ```suggestion
```http
```

POST /endorsement-provisioning/v1/activate
Host: veraison.example
Content-Type: application/vnd.veraison.elm-v1+cbor

-- body in EDN

{
  / rim-id selector / 3: [
    [ / corim / 2, / CoRIM id / "corim-acme-gizmo-1.0.0" ],
    [ / corim / 2, / CoRIM id / "corim-acme-gizmo-1.2.0" ],
    [ / comid / 0, / CoMID id / "d7bd6c2c-4844-48e4-a794-eeafdff128df" ]
  ]
}
```

#### Response

> ```

Reviewer: ```suggestion
```http
```

    "expiry": "2030-10-12T07:20:50.52Z"
  }
```

# Endorsement Lifecycle Management (ELM) Interface

This interface can be used for activating/deactivating endorsements provisioned
to the verifier. It allows using CoSERV base types to define an ELM query that 
can be used to select endorsements using either (stateful) environments or one or
more RIM identifiers.

## Query Format

Following is the CDDL format for an ELM query:
```
elm-query =  {

>   &(profile: 0) => coserv.profile

Reviewer: The profile does not belong in the rim query.  I think it makes more sense to move it into the environment query.

Reviewer 2: I used this diagnostic notation as a reference: [rv-rim-query.diag](https://github.com/ietf-rats-wg/draft-ietf-rats-coserv/blob/main/cddl/examples/rv-rim-query.diag), where the profile is included in the rim query. I'm not sure why it makes sense there but not here.
Slightly on the same topic, I would like to confirm if it's okay for the ELM query to diverge from the CoSERV data model beyond the fields that it imports? For example, I have removed `coserv.result-type` from the query because it is not useful here. Removing the profile from the rim query would be a step in the same direction.

Reviewer: > I used this diagnostic notation as a reference: [rv-rim-query.diag](https://github.com/ietf-rats-wg/draft-ietf-rats-coserv/blob/main/cddl/examples/rv-rim-query.diag), where the profile is included in the rim query. I'm not sure why it makes sense there but not here.

here, we are not constrained by the wrapping choices made by coserv, we have a bit more leeway to rearrange the objects in the most logical way for the use case… which segways into:

> Slightly on the same topic, I would like to confirm if it's okay for the ELM query to diverge from the CoSERV data model beyond the fields that it imports? 

Yes.

> For example, I have removed `coserv.result-type` from the query because it is not useful here. Removing the profile from the rim query would be a step in the same direction.

Yes.

Author: FYI: @thomas-fossati so then because we have can diverge from the CoSERV data model please note in implementation we might have to write new structs and probably modify current coserv code so that these fields can be made optional.
